core/src/main/java/org/jboss/pnc/build/finder/core/LocalFile.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.build.finder.core;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.SerializeWith;
+import org.infinispan.commons.util.Util;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+@JsonDeserialize
+@SerializeWith(LocalFile.LocalFileExternalizer.class)
+public class LocalFile {
+    private final String filename;
+
+    private final long size;
+
+    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
```

Comment:
I just realized I don't need to have `@JsonProperty` annotations there.

Out of context question. @dwalluck Where do you serialize KojiLocalArchive into JSON?

Reply:
Actually, right, you shouldn't need `@JsonCreator` or `@JsonProperty` unless the property name is different from the field name.

@michalovjan I believe that `KojiLocalAchive` is never stored since it's recreated on the fly, if that's why you're asking. For example, in `KojiBuild` I have:

```
private transient List<KojiLocalArchive> archives;
```

where the `transient` is just telling Jackson not to serialize it as it will be recreated from other preexisting information.

Reply:
If that's the case then, It's weird that there are `@JsonProperty` in KojiLocalArchive.

On the `@JsonCreator` note. Jackson was complaining for both variations without `@JsonProperty` and with `@JsonProperty` on the properties themselves. I think it needs to be explicitly on constructor because the class is immutable and doesn't have setters.

Reply:
I want to test this myself. If nothing else, maybe I'll learn something new about how Jackson works.